mbedtls/src/cipher/mod.rs

```diff
@@ -17,13 +17,15 @@ pub trait Operation: Sized {
     fn is_encrypt() -> bool;
 }
 
+#[derive(Serialize, Deserialize)]
```

Comment:
Why is this needed?

The `Encryption` and `Decryption` types are indistinguishable in serialized form, e.g. in JSON they both serialize to `{}`.